File: 백준/Silver/11659. 구간 합 구하기 4/구간 합 구하기 4.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;
import java.util.stream.Collector;
import java.util.stream.Collectors;
import java.util.stream.IntStream;

class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        StringTokenizer st = new StringTokenizer(br.readLine(), " ");
        int N = Integer.parseInt(st.nextToken());
        int M = Integer.parseInt(st.nextToken());
        int[] arr = new int[N];


        st = new StringTokenizer(br.readLine(), " ");
        for (int i = 0; i < N; i++) {
            arr[i] = Integer.parseInt(st.nextToken());
        }
//        Arrays.stream(arr).forEach(System.out::println);

            ArrayList<Integer> prefixSum = new ArrayList<>();
            prefixSum.add(0);
            int temp = 0;
            for (int i : arr) {
                temp += i;
                prefixSum.add(temp);
            }
        while (M-- > 0) {
            st = new StringTokenizer(br.readLine(), " ");
            int a = Integer.parseInt(st.nextToken());
            int b = Integer.parseInt(st.nextToken());



//        prefixSum.parallelStream().forEach(System.out::println);

            sb.append(prefixSum.get(b) - prefixSum.get(a - 1)).append("\n");
        }

        System.out.println(sb);
    }
}

File: 백준/Silver/11659. 구간 합 구하기 4/구간 합 구하기 4.py
import sys
input = sys.stdin.readline

N, M = map(int, input().split())

nums = list(map(int, input().split()))

temp = 0
prefix_sum = [0]  # 인덱스 1부터 시작하게 0

for i in nums:
    temp += i
    prefix_sum.append(temp)

# print(nums)
# print(prefix_sum)

for _ in range(M):
    i, j = map(int, input().split())
    print(prefix_sum[j] - prefix_sum[i - 1])




